app/Domain/Experience/ExperienceAggregateRoot.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace App\Domain\Experience;
+
+use App\Domain\Achievements\Experience\ExperienceAchievementUnlocker;
+use App\Domain\Achievements\PullRequest\PullRequestAchievementUnlocker;
+use App\Domain\Achievements\Series\SeriesCompletionAchievementUnlocker;
+use App\Domain\Experience\Commands\AddExperience;
+use App\Domain\Experience\Commands\RegisterPullRequest;
+use App\Domain\Experience\Commands\RegisterVideoCompletion;
+use App\Domain\Experience\Commands\UnlockAchievement;
+use App\Domain\Experience\Events\AchievementUnlocked;
+use App\Domain\Experience\Events\ExperienceEarned;
+use App\Domain\Experience\Events\PullRequestMerged;
+use App\Domain\Experience\Events\SeriesCompleted;
+use App\Domain\Experience\Events\VideoCompleted;
+use App\Models\Series;
+use App\Models\User;
+use App\Models\Video;
+use Spatie\EventSourcing\AggregateRoots\AggregateRoot;
+
+class ExperienceAggregateRoot extends AggregateRoot
+{
+    private int $experienceCount = 0;
+
+    private int $pullRequestCount = 0;
+
+    private PullRequestAchievementUnlocker $pullRequestAchievementUnlocker;
+
+    private ExperienceAchievementUnlocker $experienceAchievementUnlocker;
+
+    private SeriesCompletionAchievementUnlocker $seriesCompletionAchievementUnlocker;
+
+    public function __construct()
+    {
+        $this->pullRequestAchievementUnlocker = new PullRequestAchievementUnlocker();
```

Comment:
Perhaps there's a better way of using these unlockers? A dedicated factory felt rather heavyweight though…